episodes/04-intro-to-visualisation.Rmd

```diff
@@ -0,0 +1,228 @@
+`viridis` package.
+
+```{r ggplot-colors-adapt}
+gapminder %>%  
+  filter(year == 2007, 
+         continent == 'Americas') %>% 
+  mutate(country = fct_reorder(country, gdpPercap )) %>%
+  ggplot(aes(x = country, y = gdpPercap, fill = lifeExp   )) + 
+  geom_col()+ 
+  coord_flip()+
+  scale_fill_viridis_c()  # _c stands for continuous scale
+
+```
+
+Maybe we don't need that much information about the life expectancy. We
+only want to know if it's below or above average.
+
+```{r ggplot-colors-discrete}
+p <-  # this time let's save the plot in an object
+  gapminder %>%  
+  filter(year == 2007 & 
+         continent == 'Americas') %>% 
+  mutate(country = fct_reorder(country, gdpPercap ),
+         lifeExpCat = if_else(lifeExp >= mean(lifeExp), 'high', 'low')) %>%
+  ggplot(aes(x = country, y = gdpPercap, fill = lifeExpCat)) + 
+  geom_col()+ 
+  coord_flip()+
+  scale_fill_manual(values = c('light blue', 'orange'))  # customize the colours of the fill aesthetic
+
+```
+
+Since we saved a plot as an object, nothing has been printed out. Just
+like with any other object in `R`, if you want to see it, you need to
+call it.
+
+```{r ggplot-call}
+p
+
+```
+
+Now we can make use of the saved object and add things to it.
+
+Let's also give it a title and name the axes:
+
+```{r ggplot-titles}
+p <- 
+  p +
+  ggtitle('GDP per capita in Americas', subtitle = 'Year 2007') +
+  xlab('Country')+
+  ylab('GDP per capita')
+
+p
+```
+
+# [Writing data](https://datacarpentry.org/r-intro-geospatial/08-writing-data/index.html)
+
+## Saving the plot
+
+Once we are happy with our plot we can save it in a format of our
+choice. Remember to save it in the dedicated folder.
+
+```{r save-plot}
+ggsave(plot = p, 
+       filename = here('fig_output','plot_americas_2007.pdf'))  # By default, ggsave() saves the last displayed plot, but you can also explicitly name the plot you want to save
+
+```
+
+### Using help documentation
+
+My saved plot is not very readable. We can see why it happened by
+exploring the help documentation. We can do that by writing directly in
+the console:
+
+```{r help, eval=FALSE}
+?ggsave
+```
+
+We can read that it uses the "size of the current graphics device". That
+would explain why our saved plots look slightly different. Feel free to
+explore the documentation to see how to adapt the size e.g. by adapting
+`width`, `height` and `units` parameter.
+
+## Saving the data
+
+Another output of your work you want to save is a cleaned data set. In
+your analysis, you can then load directly that data set. Let's say we want to
+save the data only for Americas:
+
+```{r writing-data}
+gapminder_amr_2007 <- gapminder %>%
+  filter(year == 2007 & continent == 'Americas') %>%
+  mutate(country_reordered = fct_reorder(country, gdpPercap ), 
+         lifeExpCat = if_else(lifeExp >= mean(lifeExp), 'high', 'low'))
+
+write.csv(gapminder_amr_2007, here('data_output', 'gapminder_americas_2007.csv'), row.names=FALSE)
+```
```

Comment:
@fcjerome, @KyriJanssen, please add a callout block with key points here. Please add them as suggestions via a comment so that we can commit the change directly.

Reply:
```suggestion
- With ggplot2, we use the '+' operator to combine plot layers and incrementally build a more complex plot.
- In the aesthetics (aes), we can allocate variables to coordinates (x and y axes) and use the `fill` argument for colouring surfaces.
- With scale_fill_viridis_c() scale_fill_manual(), we can assign new colours to our plot.
- To open the help documentation for a function, we run the  ‘?’ sign followed by the name of the function.
```

Reply:
Thanks! I edited the text and formatting a bit.